source/DSCResources/DSC_CMCollections/DSC_CMCollections.psm1

```diff
@@ -0,0 +1,685 @@
+        Limiting collection is not evaluated in Test and is only used
+        if the collection needs created.
+
+    .PARAMETER Comment
+        Specifies a comment for the collection.
+
+    .PARAMETER RefreshSchedule
+        Specifies a schedule that determines when Configuration Manager refreshes the collection.
+
+    .PARAMETER RefreshType
+        Specifies how Configuration Manager refreshes the collection.
+        Valid values are: Manual, Periodic, Continuous, and Both.
+
+    .PARAMETER QueryRules
+        Specifies the name of the Rule and the query expression that Configuration Manager uses to update collections.
+
+    .PARAMETER ExcludeMembership
+        Specifies the collection name to exclude members from. If clients are in the excluded collection they will
+        not be added to the collection.
+
+    .PARAMETER DirectMembership
+        Specifies the resourceid for the direct membership rule.
+
+    .PARAMETER Ensure
+        Specifies if the collection is to be present or absent.
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $CollectionName,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('User','Device')]
+        [String]
+        $CollectionType,
+
+        [Parameter()]
+        [String]
+        $LimitingCollectionName,
+
+        [Parameter()]
+        [String]
+        $Comment,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $RefreshSchedule,
+
+        [Parameter()]
+        [ValidateSet('None','Periodic','Continuous','Both')]
+        [String]
+        $RefreshType,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $QueryRules,
+
+        [Parameter()]
+        [String[]]
+        $ExcludeMembership,
+
+        [Parameter()]
+        [String[]]
+        $DirectMembership,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule
+    Set-Location -Path "$($SiteCode):\"
+
+    $collection = Get-CMCollection -Name $CollectionName
+
+    if ($collection)
+    {
+        $refresh = switch ($($collection.RefreshType))
+        {
+            '1' { 'None' }
+            '2' { 'Periodic' }
+            '4' { 'Continuous' }
+            '6' { 'Both' }
+        }
+
+        $type = switch ($($collection.CollectionType))
+        {
+            '1' { 'User' }
+            '2' { 'Device' }
+        }
+
+        if ($type -eq 'User')
+        {
+            $rules = Get-CMUserCollectionQueryMembershipRule -CollectionName $collection.Name
+
+            if ($rules)
```

Comment:
This could be:
```powershell
        if ($type -eq 'User')
        {
            $rules = Get-CMUserCollectionQueryMembershipRule -CollectionName $collection.Name
            $excludes = (Get-CMUserCollectionExcludeMembershipRule -CollectionName $collection.Name).RuleName
            $directMember = (Get-CMUserCollectionDirectMembershipRule -CollectionName $collection.Name).ResourceID
        }
        else
        {
            $rules = Get-CMDeviceCollectionQueryMembershipRule -CollectionName $collection.Name
            $excludes = (Get-CMDeviceCollectionExcludeMembershipRule -CollectionName $collection.Name).RuleName
            $directMember = (Get-CMDeviceCollectionDirectMembershipRule -CollectionName $collection.Name).ResourceID
        }

        if ($rules)
        {
            foreach ($rule in $rules)
            {
                [array]$qRules += ($rule | Select-Object -Property QueryExpression, RuleName)
            }
        }
```